flang/lib/Lower/OpenMP/ClauseProcessor.cpp

```diff
@@ -953,6 +954,22 @@ bool ClauseProcessor::processMap(
           if (origSymbol && fir::isTypeWithDescriptor(origSymbol.getType()))
             symAddr = origSymbol;
 
+          if (object.sym()->owner().IsDerivedType()) {
+            omp::ObjectList objectList = gatherObjects(object, semaCtx);
+            parentObj = objectList[0];
+            parentMemberIndices.insert({parentObj.value(), {}});
```

Comment:
What happens if multiple `map` clauses have the same `parentObj`? Doesn't that overwrite the member data we had before?